### PR TITLE
IMB-87-89: Add CEPR/DOB verification and set 'duty to remove alert' value in session

### PR DESCRIPTION
**Pull request by mislam987**
## What
- Add CEPR/DOB verification to the beginning of the IMA form [IMB-87](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-87)
- Change configuration of the database, replacing 'uan' with 'cepr', and adding the new column 'duty-to-remove-alert' [IMB-88](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-88)
- Set the 'Duty to Remove alert' value in the session [IMB-89](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-89)

## Why
- To verify that the combination of CEPR and DOB exists
- To use the CEPR now instead of UAN (as part of the wider database changes to replace UAN with CEPR)
- To allow the ban-only/DTR forking in the form [IMB-85](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-85), as designed in Figma

## How
- Modify validation to cater for CEPR (6-10 digits) in validation.json
- Replace instances of 'uan' with 'cepr' across the code
- Set the 'duty to remove' value in the session across the form (/behaviours/save-form-session.js, /behaviours/validate-case-details.js, /db/save-token.js)

## Testing
- Tests passing locally

### CODE REVIEW

File: apps/verify/behaviours/validate-case-details.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-    req.sessionModel.set('service', 'ima');
-    return imaResult;
+      if (!ceprRecords.length || (ceprRecords.length && dateOfBirth.toString() !==
+        moment(req.form.values['date-of-birth']).format('YYYY-MM-DD'))) {
+        return res.redirect('/details-not-found');
+      }
+    } catch (e) {
+      return next(e);
+    }
+    return super.validate(req, res, next);
```
Comment (jamiecarterHO) [inline apps/verify/behaviours/validate-case-details.js:21]: As discussed, sticking with validation from the JSON until the new table is ready to poll for correct values.

**Review by jamiecarterHO — COMMENTED** [review]
Comments as discussed earlier

File: apps/ima/behaviours/save-form-session.js [hunk 2]
```diff
@@ -38,16 +38,17 @@ module.exports = superclass => class extends superclass {
 
       const id = req.sessionModel.get('id');
       const email = req.sessionModel.get('user-email');
-      const uan = req.sessionModel.get('uan');
+      const CEPR = req.sessionModel.get('cepr');
```
[the remark this answers is not in the document]
Comment (mislam987) [reply apps/ima/behaviours/save-form-session.js]: Changed it to lower case now to keep it consistent.

File: apps/ima/behaviours/save-form-session.js [hunk 3]
```diff
@@ -38,16 +38,17 @@ module.exports = superclass => class extends superclass {
...
+      const duty_to_remove_alert = req.sessionModel.get('duty-to-remove-alert');
 
       req.log('info', `Saving Form Session: ${id}`);
 
       try {
         const response = await axios({
           url: id ? applicationsUrl + `/${id}` : applicationsUrl,
           method: id ? 'PATCH' : 'POST',
-          data: id ? { session } : { session, email, uan, date_of_birth }
+          data: id ? { session } : { session, email, CEPR, date_of_birth, duty_to_remove_alert }
```
[the remark this answers is not in the document]
Comment (mislam987) [reply apps/ima/behaviours/save-form-session.js]: Changed it to lower case.

File: apps/verify/behaviours/validate-case-details.js (hunk 1 above)
Comment (mislam987) [reply apps/verify/behaviours/validate-case-details.js]: Changed it back to use the JSON for validation.

**Pull request merged**